debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ValueTypeHelper.java

```diff
@@ -175,13 +176,14 @@ private J9ClassPointer findJ9ClassInFlattenedClassCacheWithFieldSigImpl(Structur
 
 		@Override
 		public J9ClassPointer[] findNestedClassHierarchy(J9ClassPointer containerClazz, String[] nestingHierarchy) throws CorruptDataException {
-			J9ClassPointer[] resultClasses = new J9ClassPointer[nestingHierarchy.length + 1];
+			boolean isFlattenedArrayClass = Pattern.matches("\\[([1-9]\\d*|0)\\]", nestingHierarchy[0]);
+			J9ClassPointer[] resultClasses = new J9ClassPointer[isFlattenedArrayClass ? nestingHierarchy.length : nestingHierarchy.length + 1];
 			J9ClassPointer clazz = containerClazz;
 			resultClasses[0] = containerClazz;
 
-			for (int i = 0; i < nestingHierarchy.length; i++) {
+			for (int i = isFlattenedArrayClass ? 1 : 0; i < nestingHierarchy.length; i++) {
```

Comment:
Here, I'm skipping the first index where nestingHeirarchy would just be `[#]` so `findJ9ClassInFlattenedClassCacheWithFieldName(clazz, nestingHierarchy[i]);` would error out altogether. I'm not sure how your suggestion helps with this. The other solution I can think of is to strip the [#] from the nestingHierarchy before calling this function to avoid this problem.

Reply:
what I mean is 

```
if nestingHierarchy[i] == [#]
    resultClasses[1] = arrayClass
    //i can start from 1, no need for ? :
```

Reply:
Correct me if I'm wrong, but that will require me to add an additional parameter to that function so I can pass in the localObject.clazz() which gives me the arrayClass which has the appropriate stride. This seems to be different from the localClazz which is given by `J9ObjectHelper.clazz(localObject);`. I'm not entirely sure why these two provide different results but it doesn't seem like `localObject.clazz()` and `J9ObjectHelper.clazz(localObject);` are interchangeable.

Reply:
> but that will require me to add an additional parameter to that

you can get arrayClass with `containerClazz->arrayClass`

Reply:
> This seems to be different from the localClazz which is given by J9ObjectHelper.clazz(localObject);. I'm not entirely sure why these two provide different results but it doesn't seem like localObject.clazz()

`J9ObjectHelper.clazz` returns `localObject.clazz()` so I'm not sure why you are seeing different results